### ISSUE TITLE
Printf on Edge and latest dependencies

### ISSUE THREAD

**Issue by cgravill** [fable-compiler/Fable#1357]
### Description

I've partially tracked down a fairly strange behaviour where using `printf `on Edge and the latest dependencies throws an error: `Invalid calling object`

The full stack trace:
```
TypeError: Invalid calling object
   at Anonymous function (http://localhost:8080/bundle.js:9859:9)
   at toConsole (http://localhost:8080/bundle.js:9764:5)
   at init (http://localhost:8080/bundle.js:9586:9)
   at Anonymous function (http://localhost:8080/bundle.js:9593:1)
   at __webpack_require__ (http://localhost:8080/bundle.js:20:12)
   at Anonymous function (http://localhost:8080/bundle.js:9561:22)
   at __webpack_require__ (http://localhost:8080/bundle.js:20:12)
   at Anonymous function (http://localhost:8080/bundle.js:673:1)
   at __webpack_require__ (http://localhost:8080/bundle.js:20:12)
   at Anonymous function (http://localhost:8080/bundle.js:63:11)
```
The strange behaviour is that this only occurs in Edge, and only occurs when you *do not* have the dev tools open. A few other variations work fine, e.g. using `sprintf` then sending it to `System.Console.WriteLine`.

I can try to narrow it down further if that would be useful?

### Repro code

Follow the http://fable.io/docs/getting-started.html (included for completeness):

```
dotnet new -i Fable.Template

dotnet new fable -n FableApp -lang F#
cd FableApp

yarn install

dotnet restore
cd src
```

then update dependencies

```
.\.paket\paket.exe update
dotnet restore
yarn upgrade --dev
yarn install
```

add

```fsharp
    try
        let viaSprintf = sprintf "sprintf"
        System.Console.WriteLine viaSprintf
        // This fails in Edge when do you do not have the devtools up
        printf "printf"
    with exn ->
        System.Console.WriteLine "Error thrown!"
        System.Console.WriteLine exn.Message
        System.Console.WriteLine exn.StackTrace
```

and run `dotnet fable yarn-start`

Load in Edge without dev tools open. The console shows:

```
 sprintf

 Error thrown!

 Invalid calling object

 TypeError: Invalid calling object
   at Anonymous function (http://localhost:8080/bundle.js:9859:9)
   at toConsole (http://localhost:8080/bundle.js:9764:5)
   at init (http://localhost:8080/bundle.js:9586:9)
   at Anonymous function (http://localhost:8080/bundle.js:9593:1)
   at __webpack_require__ (http://localhost:8080/bundle.js:20:12)
   at Anonymous function (http://localhost:8080/bundle.js:9561:22)
   at __webpack_require__ (http://localhost:8080/bundle.js:20:12)
   at Anonymous function (http://localhost:8080/bundle.js:673:1)
   at __webpack_require__ (http://localhost:8080/bundle.js:20:12)
   at Anonymous function (http://localhost:8080/bundle.js:63:11)
```

### Expected and actual results

```
 sprintf
 printf
```

This occurs on FireFox, Chrome, Edge with devtools

Edge also works fine before updating dependencies. I haven't isolated which dependency is causing the issue.

### Related information

* Fable version (`dotnet fable --version`): 1.3.10
* Operating system: Windows 10
* Edge version: Microsoft Edge 41.16299.248.0

**Comment by cgravill** [comment]
I probed for the change in behaviour, it looks like it's between:

clitool dotnet-fable = 1.3.2
nuget Fable.Compiler = 1.3.2
nuget Fable.Core = 1.3.7

and

clitool dotnet-fable = 1.3.2
nuget Fable.Compiler = 1.3.2
nuget Fable.Core = 1.3.8

There were some improvements to \*printf\* functionality there so I guess that's where the trigger for the errors in Edge came in.

**Comment by alfonsogarciacaro** [comment]
Thanks a lot for taking the time to narrow down the issue @cgravill! I followed your detailed instructions and I was able to reproduce it. And thanks also to the fact you pinned the version causing the error I could quickly find this [suspicious commit](https://github.com/fable-compiler/Fable/commit/9f53009eb0638fe6eac06fdf200dfd2887235408). I tried adding the lambdas back and it seems this removes the error. It's known that some browser APIs fail when passed as values (see #715) but it's weird this only happens with devtools closed (although it's probably because the JS engine uses more optimizations in that case). Funny enough, this works well in IE 11 :)

I'll revert the commit :+1: But should we report this to the Edge team anyways (I read somewhere than different behavior than Chrome should be considered a bug)?
